imi/src/main/java/org/motechproject/nms/imi/service/impl/CsrValidatorServiceImpl.java

```diff
@@ -0,0 +1,181 @@
+    }
+
+
+    private void validateContentFileName(String pack, String fileName) {
+        if (contentFileNames == null) { loadPacks(); }
+        if (contentFileNames.containsKey(pack)) {
+            if (contentFileNames.get(pack).contains(fileName)) {
+                return;
+            }
+            throw new InvalidCsrException(String.format("Invalid contentFileName for %s pack: %s", pack, fileName));
+        }
+        throw new IllegalStateException(String.format("Unexpected pack name: %s", pack));
+    }
+
+
+    private void validateCircle(String name) {
+        if (circles == null) { loadCircles(); }
+        if (circles.contains(name)) {
+            return;
+        }
+        throw new InvalidCsrException(String.format("Invalid circle: %s", name));
+    }
+
+
+    private void validateLanguageLocationCode(String code) {
+        if (languageLocationCodes == null) { loadLanguageLocationCodes(); }
+        if (languageLocationCodes.contains(code)) {
+            return;
+        }
+        throw new InvalidCsrException(String.format("Invalid language location code: %s", code));
+    }
+
+
+    //todo: IT
+    @Override
+    public void validateSummaryRecord(CallSummaryRecordDto record) {
+        Subscription sub = subscriptionService.getSubscription(record.getRequestId().getSubscriptionId());
+        if (sub == null) {
+            throw new InvalidCsrException(String.format("Subscription %s does not exist in the database",
+                    record.getRequestId().getSubscriptionId()));
+        }
+
+        String pack = sub.getSubscriptionPack().getName();
+        validateWeekId(pack, record.getWeekId());
+        validateContentFileName(pack, record.getContentFileName());
+        validateCircle(record.getCircle());
+        validateLanguageLocationCode(record.getLanguageLocationCode());
+    }
+
+
+    private int getMaxErrorCount() {
+        try {
+            return Integer.parseInt(settingsFacade.getProperty(MAX_CDR_ERROR_COUNT));
+        } catch (NumberFormatException e) {
+            return MAX_CDR_ERROR_COUNT_DEFAULT;
+        }
+    }
+
+
+    //todo: IT
+    @Override
+    public List<String> validateSummaryRecords(Map<String, CallSummaryRecordDto> records) {
+        int maxErrorCount = getMaxErrorCount();
+        int errorCount = 0;
+        List<String> errors = new ArrayList<>();
+
+        Iterator it = records.entrySet().iterator();
+        while (it.hasNext()) {
+            Map.Entry entry = (Map.Entry) it.next();
+            try {
+                validateSummaryRecord((CallSummaryRecordDto) entry.getValue());
+            } catch (InvalidCsrException e) {
+                errors.add(e.getMessage());
+
+            }
+            if (errorCount >= maxErrorCount) {
+                errors.add(String.format("The maximum number of allowed errors of %d has been reached, " +
+                        "discarding all remaining errors.", maxErrorCount));
```

Comment:
Instead of 'discarding all remaining errors' it is stopping the processing of the file.